code/game/objects/machinery/cloning/cloning.dm

```diff
@@ -0,0 +1,245 @@
+			return
+
+		// Check if the beaker contains anything other than biomass juice
+		for(var/r in I.reagents.reagent_list)
+			var/datum/reagent/R = r
+			if(!istype(R, /datum/reagent/medicine/biomass) && !istype(R, /datum/reagent/medicine/biomass/xeno))
+				to_chat(user, "<span class='warning'>\The [src] rejects the beaker.</span>")
+				return
+
+		beaker = I
+		if(!user.transferItemToLoc(I, src))
+			return
+
+		update_icon()
+
+		return TRUE
+
+
+/obj/machinery/cloning/vats/examine(mob/user)
+	. = ..()
+	if(!beaker)
+		to_chat(user, "<span class='notice'>It doesn't have a beaker attached.</span>")
+		return
+	if(timerid)
+		to_chat(user, "<span class='notice'>There is something weird inside.</span>")
+		return
+	if(occupant)
+		to_chat(user, "<span class='notice'>It looks like there is a human in there!</span>")
+		return
+
+
+/obj/machinery/cloning/vats/update_icon()
+	if(!beaker)
+		icon_state = "cell_0"
+		return
+	if(occupant || timerid)
+		icon_state = "cell_growing"
+		return
+	var/amount = clamp(round(beaker.reagents.total_volume / biomass_required, 25), 0, 100)
+	icon_state = "cell_[amount]"
+
+
+/obj/machinery/cloning/vats/proc/grow_human(instant = FALSE)
+	// Ensure we cleanup the beaker contents
+	if(beaker)
+		beaker.reagents.remove_all(biomass_required)
+
+	if(instant)
+		finish_growing_human()
+		return
+
+	timerid = addtimer(CALLBACK(src, .proc/finish_growing_human), grow_timer, TIMER_STOPPABLE)
+	update_icon()
+
+
+/obj/machinery/cloning/vats/proc/eject_user(silent = FALSE)
+	if(!occupant)
+		return
+
+	if(!silent)
+		visible_message("[icon2html(src, viewers(src))] <span class='notice'><b>[src]</b> ejects the freshly spawned clone.</span>")
+	occupant.forceMove(get_step(loc, dir))
+	occupant.Paralyze(10 SECONDS)
+	occupant.disabilities &= ~(BLIND | DEAF)
+	occupant.set_blindness(10, TRUE)
+	to_chat(occupant, {"
+<span class='notice'>You are a frestly spawned clone, you appear as a Squad marine, but nothing more.
+You remember nothing of your past life.
+
+You are weak, best rest up and get your strength before fighting.</span>"})
+
+	occupant = null
+	update_icon()
+
+
+/obj/machinery/cloning/vats/proc/finish_growing_human()
+	occupant = new(src)
+	var/datum/job/J = SSjob.GetJobType(/datum/job/terragov/squad/vatgrown)
+	occupant.apply_assigned_role_to_spawn(J)
+	occupant.set_species("Early Vat-Grown Human")
+	occupant.fully_replace_character_name(occupant.real_name, occupant.species.random_name(occupant.gender))
+	occupant.disabilities |= (BLIND & DEAF)
+	occupant.set_blindness(10) // Temp fix until blindness is fixed.
+	// Blindness doenst't trigger with just the disability, you need to set_blindness
+
+	GLOB.offered_mob_list += occupant
+	notify_ghosts("<span class='boldnotice'>A new clone is available! Name: [name]</span>", enter_link = "claim=[REF(occupant)]", source = src, action = NOTIFY_ORBIT)
+
+	// Cleanup the timers
+	deltimer(timerid)
```

Comment:
```suggestion
```
This is triggered either when no timer exists or when the timer is finished. No need to delete it, just null the reference.

Reply:
It was mostly incase admemes